### DISCUSSION TITLE
dash-case props in components?

### DISCUSSION THREAD

**Discussion by s1n7ax** [leptos-rs/leptos#2588] [Q&A] [answered]
In react.js we can have dash-case attributes in components using aliases. I can see `web_sys::HtmlElement` can have dash-case attributes like `aria-label`. Just wondering if leptos elements can define dash-case props?

```javascript
const Link = ({
  to,
  className,
  onClick,
  target,
  rel,
  key,
  'aria-label': ariaLabel,
}) => {
};
```

**Comment by gbj** [comment, marked as answer]
Props are just a struct, so each prop is a struct field name. Rust identifiers can't have dashes (as far as I know) so struct field names/prop names can't have dashes. 

I guess in theory it's possible to modify the view macro so that it rewrites any dash-cased prop names to snake_case but it just seems like an extra layer of confusion to me.

**Reply by s1n7ax** [reply]
well, we can make props optional by using `#[prop(optional)]` so shouldn't it be possible to set the prop name by setting something like `#[prop(name="aria-label")] aria_label: String`? I mean technically possible right?

**Reply by gbj** [reply]
Not really — the `view` macro and the `component` macro don't have any knowledge of one another's internals (and can't), so if you use `<Something aria-label="something"/>` in a view, it has no way to know that this should expand to `Something(SomethingProps::builder().aria_label("something"))` (which is effectively what the view macro does).
